Sources/WeaverDI/DependencyContainer.swift

```diff
@@ -16,8 +16,8 @@ open class DependencyContainer {
     private let parent: DependencyContainer?
     
     init(parent: DependencyContainer? = nil,
-         builderStore: BuilderStoring = BuilderStore(),
-         instanceCache: InstanceCaching = InstanceCache()) {
+         builderStore: BuilderStoring = ThreadSafeBuilderStoringDecorator(builderStoring: BuilderStore()),
```

Comment:
Let's keep it the way it was and wrap the `instanceCache` & `builderStore` internally like so:
```swift
public init(_ parent: DependencyContainer? = nil,
                 builderStore: BuilderStoring = BuilderStore(),
                 instanceCache: InstanceCaching = InstanceCache()) {
    self.parent = parent
    instances = ThreadSafeInstanceCachingDecorator(cache: InstanceCache())
    builders = ThreadSafeBuilderStoringDecorator(builderStoring: BuilderStore())

    ...
}
```

I think this is better because we never want users to mistakenly build a non thread safe `DependencyContainer`.

Reply:
Done